cephci/utils/configure.py

```diff
@@ -76,3 +80,180 @@ def setup_ssh_keys(installer, nodes):
+    if out:
+        log.error("Failed to update-ca-trust")
+        return False
+    return True
+
+
+def validate_trusted_list(node):
+    """Verify trust list is updated with hostname"""
+    cmd = f"trust list | grep -i {node.hostname}"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if node.hostname not in out:
+        log.error("Trust list not updated")
+        return False
+    return True
+
+
+def copy_cert_to_secondary_node(node, secondary_node):
+    """Copy the certificate to anotheer node in the cluster"""
+    cmd = f"scp /opt/registry/certs/domain.crt root@{secondary_node[1].hostname}:/etc/pki/ca-trust/source/anchors/"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to copy the certificate to another node")
+        return False
+    return True
+
+
+def start_local_private_registry(node):
+    """Start local private registry"""
+    cmd = """podman run --restart=always --name myprivateregistry \
+    -p 5000:5000 -v /opt/registry/data:/var/lib/registry:z \
+    -v /opt/registry/auth:/auth:z \
+    -v /opt/registry/certs:/certs:z \
+    -e "REGISTRY_AUTH=htpasswd" \
+    -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+    -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+    -e "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt" \
+    -e "REGISTRY_HTTP_TLS_KEY=/certs/domain.key" \
+    -e REGISTRY_COMPATIBILITY_SCHEMA1_ENABLED=true \
+    -d registry:2"""
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    if out:
+        log.error("Failed to start local secure private registry")
+        return False
+
+
+def add_images_to_private_registry(
+    node, registry_uname, registry_passwd, registry, build_type
+):
+    """Red Hat Ceph Storage 5 image, Prometheus images, and Dashboard image from the Red Hat Customer Portal
+    to the private registry:"""
+
+    base_cmd = (
+        "podman run -v /opt/registry/certs:/certs:Z -v /opt/registry/certs/domain.cert:/certs/domain.cert:Z "
+        "--rm registry.redhat.io/rhel8/skopeo:8.5-8 skopeo copy  "
+        "--remove-signatures --src-creds myusername:mypassword1 "
+        "--dest-cert-dir=./certs/ "
+        f"--dest-creds {registry_uname}:{registry_passwd}"
+    )
+
+    # Get the required images here
+    images = get_images(build_type)
+    for image in images:
+        cmd = f"{base_cmd} docker://{registry}/{image} docker://{node.hostname}:5000/{image}"
+        out, _ = node.exec_command(cmd=cmd, sudo=True)
+        if out:
+            log.error(f"Failed to copy {image} to private registry")
+            return False
+    return True
+
+
+def accept_eula(node):
+    cmd = "cat" if accept_eula else "touch"
+    cmd += " /usr/share/ibm-storage-ceph-license/accept"
+
+    Package(node).install("ibm-storage-ceph-license", env_vars=accept_eula)
+    return node.exec_command(sudo=True, cmd=cmd, check_ec=True)
+
+
+def exec_cephadm_preflight(installer, build_type, ceph_repo):
```

Comment:
As preflight playbook is not part of disconnected install better to send separate PR